apps/gyrokinetic.c

```diff
@@ -873,27 +878,90 @@ gkyl_gyrokinetic_app_write_rad_emissivity(gkyl_gyrokinetic_app* app, int sidx, d
   );
   
   struct gk_species *s = &app->species[sidx];
+  const struct gkyl_array *fin_neut[app->num_neut_species];
   const struct gkyl_array *fin[app->num_species];
   for (int i=0; i<app->num_species; ++i) 
     fin[i] = app->species[i].f;
-  gk_species_radiation_emissivity(app, s, &s->rad, fin);
+  for (int i=0; i<app->num_neut_species; ++i)
+    fin_neut[i] = app->neut_species[i].f;
+
+  gk_species_radiation_emissivity(app, s, &s->rad, fin, fin_neut);
   for (int i=0; i<s->rad.num_cross_collisions; i++) {
     // copy data from device to host before writing it out
     if (app->use_gpu) {
       gkyl_array_copy(s->rad.emissivity_host[i], s->rad.emissivity[i]);
     }
     // Construct the file handles for vparallel and mu drag
-    const char *fmt_emissivity = "%s-%s_emissivity_%s_%d.gkyl";
-    int sz_emissivity = gkyl_calc_strlen(fmt_emissivity, app->name, s->info.name, app->species[s->rad.collide_with_idx[i]].info.name, frame);
-    char fileNm_emissivity[sz_emissivity+1]; // ensures no buffer overflow
-    snprintf(fileNm_emissivity, sizeof fileNm_emissivity, fmt_emissivity, app->name,
-      s->info.name, app->species[s->rad.collide_with_idx[i]].info.name, frame);
-    gkyl_comm_array_write(s->comm, &app->grid, &app->local, mt, s->rad.emissivity_host[i], fileNm_emissivity);
+    const char *fmt_emissivity = "%s-%s_emissivity_%s_%d.gkyl";  
+    if (s->rad.is_neut_species[i]) {
+      int sz_emissivity = gkyl_calc_strlen(fmt_emissivity, app->name, s->info.name, app->neut_species[s->rad.collide_with_idx[i]].info.name, frame);
+      char fileNm_emissivity[sz_emissivity+1]; // ensures no buffer overflow
+      snprintf(fileNm_emissivity, sizeof fileNm_emissivity, fmt_emissivity, app->name, s->info.name, app->neut_species[s->rad.collide_with_idx[i]].info.name, frame);
+      gkyl_comm_array_write(s->comm, &app->grid, &app->local, mt, s->rad.emissivity_host[i], fileNm_emissivity);
+    } else {
+      int sz_emissivity = gkyl_calc_strlen(fmt_emissivity, app->name, s->info.name, app->species[s->rad.collide_with_idx[i]].info.name, frame);
+      char fileNm_emissivity[sz_emissivity+1]; // ensures no buffer overflow
+      snprintf(fileNm_emissivity, sizeof fileNm_emissivity, fmt_emissivity, app->name, s->info.name, app->species[s->rad.collide_with_idx[i]].info.name, frame);
+      gkyl_comm_array_write(s->comm, &app->grid, &app->local, mt, s->rad.emissivity_host[i], fileNm_emissivity);
```

Comment:
@JunoRavin Should this comm_array_write use the species communicator or app communicator? Are they different?

I had issues reading the files with pgkyl when the species communicator was used here (multi-gpu simulation). I switched this to app->comm and the files seemed to be written correctly.

Reply:
If an output is on the configuration space grid, it should be app->comm. If an output is on the phase space grid, it should be species->comm.